clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -3506,6 +3538,68 @@ protected Set<String> maybeSendIngestionHeartbeat() {
     return failedPartitions;
   }
 
+  /**
+   *  Resubscribe operation by passing new version role and partition role to {@link AggKafkaConsumerService}. The action
+   *  for leader and follower replica will be handled differently.
+   */
+  private void resubscribe(
+      PartitionConsumptionState partitionConsumptionState,
+      TopicPartitionReplicaRole.VersionRole versionRoleFromConsumerAction) throws InterruptedException {
+    if (isLeader(partitionConsumptionState)) {
+      resubscribeAsLeader(partitionConsumptionState, versionRoleFromConsumerAction);
+    } else {
+      resubscribeAsFollower(partitionConsumptionState, versionRoleFromConsumerAction);
+    }
+  }
+
+  protected void resubscribeAsFollower(
+      PartitionConsumptionState partitionConsumptionState,
+      TopicPartitionReplicaRole.VersionRole versionRoleFromConsumerAction) throws InterruptedException {
+    int partition = partitionConsumptionState.getPartition();
+    consumerUnSubscribe(versionTopic, partitionConsumptionState);
+    waitForAllMessageToBeProcessedFromTopicPartition(
```

Comment:
Timeout exception might happen here, which is not fatal previously when server does unsubscription for standby to offline transition or when server does graceful shutdown; however, I don't think exception is okay during resubscription, right?
Let's think about it thoroughly; if unsubscription fails somehow, regardless of the error types, do we fail this partition or continue with the subscription in the new pool?

Reply:
@lluwm Could you please help check if we need to set the unsubscribe flag in PCS here?

Reply:
Actually, we also did similar thing during `LEADER_TO_STANDBY` consumer action processing. It basically did the same thing to unsub and then subscribe to the different topic, and it is also executed by SIT thread. Fail logic should be similar here, we fail the partition and this version on this host.